dashboard/app/controllers/activities_controller.rb

```diff
@@ -160,16 +162,27 @@ def track_progress_for_user
     else
       @activity = Activity.create_async!(attributes)
     end
-
     if @script_level
-      @new_level_completed = current_user.track_level_progress_async(
-        script_level: @script_level,
-        new_result: test_result,
-        submitted: params[:submitted] == "true",
-        level_source_id: @level_source.try(:id),
-        level: @level,
-        pairing_user_ids: pairing_user_ids
-      )
+      if synchronous_save
```

Comment:
This looks fine, but is there a particular motivation  for adding the sync/async saving here? Is this related to reducing dependence on Activity, or is it just while you're in here? If my understanding is correct, track_user_level_progress_async will turn into sync unless gatekeeper flag is set to allow async. Which is... never? So does this sync/async split change anything in practice?